diff --git a/arch/arm/mach-bcm2708/bcm2708.c b/arch/arm/mach-bcm2708/bcm2708.c
index caa9f48..668905d 100644
--- a/arch/arm/mach-bcm2708/bcm2708.c
+++ b/arch/arm/mach-bcm2708/bcm2708.c
@@ -140,7 +140,12 @@ static struct map_desc bcm2708_io_desc[] __initdata = {
 	 .virtual = IO_ADDRESS(GPIO_BASE),
 	 .pfn = __phys_to_pfn(GPIO_BASE),
 	 .length = SZ_4K,
-	 .type = MT_DEVICE}
+	 .type = MT_DEVICE},
+	{
+         .virtual = IO_ADDRESS(V3D_BASE),
+         .pfn = __phys_to_pfn(V3D_BASE),
+         .length = SZ_4K,
+         .type = MT_DEVICE}
 };
 
 void __init bcm2708_map_io(void)
diff --git a/arch/arm/mach-bcm2708/include/mach/platform.h b/arch/arm/mach-bcm2708/include/mach/platform.h
index 2e7e1bb..f0df6b3 100644
--- a/arch/arm/mach-bcm2708/include/mach/platform.h
+++ b/arch/arm/mach-bcm2708/include/mach/platform.h
@@ -77,6 +77,8 @@
 #define USB_BASE                 (BCM2708_PERI_BASE + 0x980000) /* DTC_OTG USB controller */
 #define MCORE_BASE               (BCM2708_PERI_BASE + 0x0000)   /* Fake frame buffer device (actually the multicore sync block*/
 
+#define V3D_BASE (BCM2708_PERI_BASE + 0xc00000)
+
 #define ARMCTRL_BASE             (ARM_BASE + 0x000)
 #define ARMCTRL_IC_BASE          (ARM_BASE + 0x200)           /* ARM interrupt controller */
 #define ARMCTRL_TIMER0_1_BASE    (ARM_BASE + 0x400)           /* Timer 0 and 1 */
